### PR TITLE
Remove reduntant text after a progressive account is onboarded

### LINKED ISSUE #8278: Remove duplicate/redundant "Ready to start Selling" text
### Describe the bug

When a user is onboarded progressively, we show modal with confetti confirming that this was successful.

However, we also have a similar message which appears on the dashboard which we can now remove because it is duplicating the modal and therefore redundant.

### To Reproduce

1. Onboard a progressive account.
2. Verify you see the success modal.
3. Once this is closed, there is a further "You're ready to start selling" card which can be removed since it's duplicating the modal.

### Screenshots

Screenshot 2024-02-27 at 16 16 42

### PR DESCRIPTION
**Pull request by dpaun1985**
Fixes #8278 

#### Changes proposed in this Pull Request

Remove reduntant message after a progressive account is onboarded.
Screenshot 2024-03-18 at 14 49 35

#### Testing instructions
1. For your current account, go to server db, and change the `account_type` field in `wcpay_accounts`  table, from `wcpay` to `wcpay_other`
2. go to WCPay Dev and clear account cache
3. Go to `Payments` and press Finish button 
4. Onboard on Stripe.
5. press the return to dev shared account
6. check if you can see the success modal
Screenshot 2024-03-18 at 14 54 43
7. The message "You're ready to start selling" shouldn't be shown amymore

https://github.com/Automattic/woocommerce-payments/assets/82826872/a49e2f26-d9ac-439e-a3a8-801056d1e1c5

*

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [ ] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8399` or branch name `fix/8278-remove-redundant-text` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/8278-remove-redundant-text) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 3a5db01cbcece354349a91511ab15e4457b7a578
- Build time: 2024-03-25 12:59:36 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/overview/connection-sucess-notice.tsx [hunk 1]
```diff
@@ -40,46 +40,29 @@ const ConnectionSuccessNotice: React.FC = () => {
...
 		);
 	};
-
-	return ! isDismissed && ! onboardingTestMode ? (
+	{
+		/* Show at the end of Progressive Onboarding or at the end of the full KYC flow. */
+	}
+	return ! isDismissed &&
+		! onboardingTestMode &&
+		! ( isPoEnabled && ! isPoComplete ) ? (
```
Comment (elazzabi) [inline client/overview/connection-sucess-notice.tsx]: This started to get a little bit too long, with many `&&` and `!` which makes reading it a little challenging 😅 . I wonder if we can group a few of these in some variable so it's easier to read.

**Review by elazzabi — CHANGES_REQUESTED** [review]
Looks good! I left a small request for change, but should be good to go after fixing it!

Thanks!

File: client/overview/connection-sucess-notice.tsx (hunk 1 above)
Comment (dpaun1985) [reply client/overview/connection-sucess-notice.tsx]: changed in https://github.com/Automattic/woocommerce-payments/pull/8399/commits/4951606efdeb39fe8fa2fc8b01dd8718276386ff. thanks @elazzabi

File: client/overview/connection-sucess-notice.tsx [hunk 2]
```diff
@@ -40,46 +40,25 @@ const ConnectionSuccessNotice: React.FC = () => {
 			/>
 		);
 	};
-
-	return ! isDismissed && ! onboardingTestMode ? (
+	const isPoDisabledOrCompleted = ! ( isPoEnabled && ! isPoComplete );
```
Comment (elazzabi) [inline client/overview/connection-sucess-notice.tsx]: Nitpicking. Given the name of the variable is disabled or completed, I suggest we do the same in the definition:
```suggestion
	const isPoDisabledOrCompleted = ! isPoEnabled || isPoComplete;
```

**Review by elazzabi — CHANGES_REQUESTED** [review]

File: changelog/fix-8278-remove-redundant-text [hunk 3]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: fix
+
+Remove redundant message after account is onboarded
```
Comment (elazzabi) [inline changelog/fix-8278-remove-redundant-text]:
```suggestion
Remove redundant message after the account is onboarded
```

Comment (dpaun1985) [reply changelog/fix-8278-remove-redundant-text]: changed in https://github.com/Automattic/woocommerce-payments/pull/8399/commits/2adf682ee866d42705654cc9ae8b1e0678d6eb02 . thx.

File: client/overview/connection-sucess-notice.tsx (hunk 2 above)
Comment (dpaun1985) [reply client/overview/connection-sucess-notice.tsx]: changed in https://github.com/Automattic/woocommerce-payments/pull/8399/commits/2adf682ee866d42705654cc9ae8b1e0678d6eb02 . thx.

**Review by elazzabi — APPROVED** [review]

**Comment by anu-rock** [comment]
A nitpicky comment - the title of the PR sounds too generic. As a developer, I am forced to go deeper into the issue or PR description to get more context, which is an extra step especially when git blaming or git bisecting.

**Pull request merged**